cognite/client/data_classes/datapoints.py

```diff
@@ -266,22 +267,21 @@ class DatapointsQuery:
     """Represent a user request for datapoints for a single time series"""
 
     _NOT_SET = object()
-
-    OPTIONAL_DICT_KEYS: ClassVar[frozenset[str]] = frozenset(
+    _API_DEFAULTS: ClassVar[MappingProxyType[str, Any]] = MappingProxyType(
         {
-            "start",
-            "end",
-            "aggregates",
-            "granularity",
-            "timezone",
-            "target_unit",
-            "target_unit_system",
-            "limit",
-            "include_outside_points",
-            "ignore_unknown_ids",
-            "include_status",
-            "ignore_bad_datapoints",
-            "treat_uncertain_as_bad",
+            "start": 0,
+            "end": "now",
+            "aggregates": None,
+            "granularity": None,
+            "timezone": None,
+            "target_unit": None,
+            "target_unit_system": None,
+            "limit": None,
+            "include_outside_points": False,
+            "ignore_unknown_ids": False,
+            "include_status": False,
+            "ignore_bad_datapoints": True,
+            "treat_uncertain_as_bad": True,
```

Comment:
I'm not quite sure here, it seems to apply defaults to a range of parameters, not all of them were added recently

- do defaults here match defaults on API? Don't see all in https://api-docs.cognite.com/20230101/tag/Time-series/operation/getMultiTimeSeriesDatapoints some are handled in SDK?
- do existing tests cover this change? (see that no tests are updated, it could be that behavior stayed the same, but could be lack of coverage)

Reply:
for 1st point I needed to expand "items" in body, there there's more parameters :+1:

Reply:
These defaults should have been used from "day one". It now exactly matches the existing `retrieve` methods (`retrieve`, `retrieve_arrays`, `retrieve_dataframe` and `*_in_tz`). 

> ...not all of them were added recently

True, the problem was SDK code relating to the new object aggregates. The API happily returned datapoints, but unpacking failed.

Reply:
So, yes, these match the API, _although we are a tiny bit more lenient_. The API doesn't allow `ignoreUnknownIds` to be customized _per time series_ that you request datapoints from, but the SDK allows this to keep all options to be set individually, not "every option except ignoreUnknownIds".

The SDK also never passes any top-level values in the payload, and instead explicitly specifies all settings (that are not using the default value) for all time series.